wp_api/src/login/url_discovery.rs

```diff
@@ -224,6 +227,36 @@ impl AutoDiscoveryAttemptResult {
             Err(_) => None,
         }
     }
+
+    fn is_local_dev_environment(&self) -> bool {
+        match &self.api_discovery_result {
+            Ok(success) => is_local_dev_environment_url(&success.parsed_site_url),
+            Err(error) => {
+                if let Some(parsed_url) = error.parsed_site_url() {
+                    is_local_dev_environment_url(parsed_url)
+                } else {
+                    false
+                }
+            }
+        }
+    }
+
+    /// Does the site look like a WordPress site?
+    fn is_wordpress_site(&self) -> bool {
+        self.is_wordpress_site.is_successful()
+    }
+}
+
+// Does the given URL look like it's on a local development environment for the purposes of the Login Spec?
+pub fn is_local_dev_environment_url(parsed_site_url: &ParsedUrl) -> bool {
```

Comment:
Is this only used by the testing suite? The implementation of this is pretty speculative, so if we need it for testing suite, I'd prefer if it's not part of the public API. I wouldn't mind even having separate implementations in Swift, Kotlin & Rust since it's so simple.

Reply:
I'm not sure if it needs to be public API at this point but it is used in two places to figure out which error message to show.

But it's an implementation detail so we can probably hide it from the public API

Reply:
I think we'll want to move the code that uses this to Rust and then we can hide it from the API. I am still worried of it being speculative, but maybe we can address that with some naming changes for both the function name and the error message.